test/parse/locator.test.js

```diff
@@ -28,76 +15,98 @@ describe('DOMLocator', () => {
+				'  <a attr="value"><![CDATA[1]]>something\n' +
+				'</a>x</test>',
 			'text/xml'
 		)
-		var test = doc.documentElement
-		var a = test.firstChild.nextSibling
-		assertPosition(doc.firstChild, 1, 1, 'first child')
-		assertPosition(
-			doc.firstChild.nextSibling,
-			1,
-			1 + '<?xml version="1.0"?>'.length,
-			'first child nextSibling'
-		)
-		assertPosition(test, 2, 1, 'document element' + test)
-		//assertPosition(test.firstChild, 1, 7);
-		assertPosition(a, 3, 3, 'documentElement firstchild nextSibling' + a)
-		assertPosition(a.firstChild, 3, 19, 'a.firstchild')
-		assertPosition(
-			a.firstChild.nextSibling,
-			3,
-			19 + '<![CDATA[1]]>'.length,
-			'a.firstchild.nextsibling'
-		)
-		assertPosition(test.lastChild, 4, 5, 'test.lastChild')
-	})
 
-	it('error positions', () => {
-		var error = []
-		var parser = new DOMParser({
-			locator: { systemId: 'c:/test/1.xml' },
-			errorHandler: function (msg) {
-				error.push(msg)
+		expect(dom).toMatchObject({
+			firstChild: {
+				// <?xml version="1.0"?>
+				lineNumber: 1,
+				columnNumber: 1,
+				nextSibling: {
+					nodeName: '#comment',
+					lineNumber: 1,
+					columnNumber: 1 + instruction.length,
+				},
+			},
+			documentElement: {
+				nodeName: 'test',
+				lineNumber: 2,
+				columnNumber: 1,
+				firstChild: {
+					nodeName: '#text',
+					lineNumber: 2,
+					columnNumber: 7,
+					nextSibling: {
+						nodeName: 'a',
+						lineNumber: 3,
+						columnNumber: 3,
+						firstChild: {
+							nodeName: '#cdata-section',
+							lineNumber: 3,
+							columnNumber: 19,
+						},
+						lastChild: {
+							textContent: 'something\n',
+							lineNumber: 3,
+							columnNumber: 32,
+						},
+					},
+				},
+				lastChild: {
+					textContent: 'x',
+					lineNumber: 4,
+					columnNumber: 5,
+				},
 			},
 		})
-		var xml = '<html><body title="1<2"><table>&lt;;test</body></body></html>'
-		var doc = parser.parseFromString(xml, 'text/html')
-		var attr = doc.documentElement.firstChild.attributes.item(0)
-		assertPosition(attr, 1, 19, 'title="1<2 ')
-		assert(error.length, 0)
 	})
 
-	it('error positions p', () => {
-		var error = []
-		var parser = new DOMParser({
-			locator: {},
-			errorHandler: function (msg) {
-				error.push('@@' + msg)
-			},
+	it('attribute position', () => {
+		// TODO: xml not well formed but no warning or error, extract into different test?
+		const xml = '<html><body title="1<2"><table>&lt;;test</body></body></html>'
+		const { errors, parser } = getTestParser({
+			locator: { systemId: 'c:/test/1.xml' },
 		})
-		var doc = parser.parseFromString('<root>\n\t<err</root>', 'text/html')
-		var root = doc.documentElement
-		var textNode = root.firstChild
-		assert.isTrue(
-			/\n@#\[line\:2,col\:2\]/.test(error.join(' ')),
-			'line,col must record:' + JSON.stringify(error)
-		)
+
+		const doc = parser.parseFromString(xml, 'text/html')
+
+		expect({ actual: doc.toString(), ...errors }).toMatchSnapshot()
```

Comment:
As described in my comment on line 87 the tests doesn't yield any errors or warnings although the XML is not well formed. I made this explicit by also asserting on no errors being present.

Reply:
Got it.

In this case, shouldn't we assert that there are no errors in some of the other tests as well?

Reply:
I'm quite confident that I did it for all cases that I thought it makes sense for some reason, but will double check and maybe add a related comment or some README ...

Reply:
I think I will keep this work as part of checking for more duplicates and potentially rearranging them better as discussed in #72 

Also since I want to get this PR done.